documentation/api/io.strimzi.api.kafka.model.connect.build.Build.adoc

```diff
@@ -0,0 +1,191 @@
+Configures additional connectors for Kafka Connect deployments.
+
+=== `output`
+
+In order to build new container images with additional connector plugins, Strimzi requires a container registry where the images can be pushed to, stored and pulled from.
+Strimzi does not run its own container registry, so users need to provide their own registry.
+Strimzi is able to use private container registries as well as public registries such as link:https://quay.io/[Quay^] or link:https://hub.docker.com//[Docker Hub^].
+The container registry is configured in the `.spec.build.output` section of the `KafkaConnect` custom resource.
+The `output` configuration supports two types: `docker` and `imagestream`.
+The `output` section is required.
+
+.Using Docker registry
+
+To use a Docker registry, you have to specify the `type` as `docker`, and the `image` field with the full name of the new container image.
+The full name has to include:
+
+* The address of the registry.
+* Port number if it is listening on non-standard port
+* The tag of the new container image
+
+Examples of valid image names are:
+
+* `docker.io/my-org/my-image/my-tag`
+* `quay.io/my-org/my-image/my-tag`
+* `image-registry.image-registry.svc:5000/myproject/kafka-connect-build:latest`
+
+Different Kafka Connect deployments should use a different image (for example at least with different tags).
+
+If the registry requires authentication, the field `pushSecret` can be used to set a name of the Secret with the registry credentials.
+The Secret should use the `kubernetes.io/dockerconfigjson` type and contain the `.dockerconfigjson` file with the Docker credentials.
+For more details see link:https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials[Create a Secret based on existing Docker credentials^]
```

Comment:
```suggestion
For more information on pulling an image from a provate registry, see link:https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials[Create a Secret based on existing Docker credentials^].
```

Reply:
Do we want this link in the attributes file? I guess it might depend on whether there's a downstream equivalent to swap. (Off topic - maybe we should look at how we manage external refs. Not sure of the necessity of using the attributes file if links don't change)

Reply:
I can move the external link to the attributes. I guess it makes sense since we do it for most other external links. But I do not think we need to replace this one for downstream. But having them in one place will make it easier to mass update them in the future for example etc.